### ISSUE TITLE
How to decrease the learning rate every 10 epochs by a factor of 0.9?

### ISSUE THREAD

**Issue by Nafees-060** [tensorflow/tensorflow#52529]
I want to set the learning rate at 10^-3 with a decay every 10 epochs by a factor of 0.9. I am using the Adam optimizer in Tensorflow Keras. I have found this code in the official documentation:

initial_learning_rate = 0.1

```
lr_schedule = tf.keras.optimizers.schedules.ExponentialDecay(
    initial_learning_rate,
    decay_steps=100000,
    decay_rate=0.96,
    staircase=True
)
```
I do not know what is this decay_steps=100000. Actually, I want to decrease my learning rate after 10 epochs. How can I do it?

**Comment by jKaushalb** [comment]
In tensorflow it uses mini batches to train the model. For your problem you need to understand two things 
1) when we uses generators on our datasets it keeps on giving batches infinitely so we need to tell keras that after these many batches we consider as one epoch in
 model.fit( steps_per_epoch) argument. so generally we uses **steps_per_epoch = train_dataset.shape[0] // batch_size** 
(total training example / batchsize)

for more info  [check this](https://stackoverflow.com/questions/49922252/choosing-number-of-steps-per-epoch)

2) here you want that after 10 epochs your lr should be lr = lr * 0.9  so in exponential decay uses this equation 
`def decayed_learning_rate(step):
  return initial_learning_rate * decay_rate ^ (step / decay_steps)`

where step/decay_steps is integer hence it will 1 when step == decay_steps other wise if step<decay_step it will be zero
so in **decay_steps you pass
 ### decay_steps = 10 * steps_per_epoch**

another approach would be you can create your own optimzer for your requirement

<pre>class MyLRSchedule(tf.keras.optimizers.schedules.LearningRateSchedule):

  def __init__(self, initial_learning_rate,decay=10000):
    self.initial_learning_rate = initial_learning_rate

  def __call__(self, step):
     return self.initial_learning_rate / (step + 1)

 '''here in call you can check conditions and customize the things you want to do like here you can add if condition to reduce your learning rate do
    if (step % decay ==0)
   {
       self.initial_learning_rate = self.initial_learning_rate * 0.9
   }
return self.initial_learning_rate '''
optimizer = tf.keras.optimizers.SGD(learning_rate=MyLRSchedule(0.1,steps_per_epoch))</pre>

links:
[example link](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/LearningRateSchedule)
[other schedules](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules)

**Comment by mohantym** [comment]
Hi @Nafees-060! Could you check with above suggestions?

**Comment by Nafees-060** [comment]
> Hi @Nafees-060! Could you check with the above suggestions?

I am working on it, please give me some time
